src/PhpSpreadsheet/Worksheet/Table.php

```diff
@@ -0,0 +1,471 @@
+<?php
+
+namespace PhpOffice\PhpSpreadsheet\Worksheet;
+
+use PhpOffice\PhpSpreadsheet\Cell\Coordinate;
+use PhpOffice\PhpSpreadsheet\Exception as PhpSpreadsheetException;
+use PhpOffice\PhpSpreadsheet\Worksheet\Table\TableStyle;
+
+class Table
+{
+    /**
+     * Table Name.
+     *
+     * @var string
+     */
+    private $name = '';
+
+    /**
+     * Show Header Row.
+     *
+     * @var bool
+     */
+    private $showHeaderRow = true;
+
+    /**
+     * Show Totals Row.
+     *
+     * @var bool
+     */
+    private $showTotalsRow = false;
+
+    /**
+     * Table Range.
+     *
+     * @var string
+     */
+    private $range = '';
+
+    /**
+     * Table Worksheet.
+     *
+     * @var null|Worksheet
+     */
+    private $workSheet;
+
+    /**
+     * Table Column.
+     *
+     * @var Table\Column[]
+     */
+    private $columns = [];
+
+    /**
+     * Table Style.
+     *
+     * @var TableStyle
+     */
+    private $style;
+
+    /**
+     * Create a new Table.
+     *
+     * @param string $range (e.g. A1:D4)
+     *
+     * @return $this
+     */
+    public function __construct(string $range = '', ?Worksheet $worksheet = null)
```

Comment:
Should the constructor also allow the table name as an argument? As it's a mandatory value for a table?

Reply:
Table name as a constructor argument is a nice addition. Also, the second argument `$worksheet` _(grabbed from autofilter)_ may get confused with `$worksheet->addTable()`. So I would try to replace it with optional `tableName`.

And the table name is not mandatory. Like the excel application, the default name will be used while writing the Xlsx file.

```php
public function __construct(string $range = '', string $name = '')
```

Reply:
Implemented 530e6642bf26522176504fccc81385d7358936d2